### ISSUE TITLE
Unable to update complex args for service

### ISSUE THREAD

**Issue by RangerMauve** [docker/docker#24849]
**Output of `docker version`:**

```
Docker version 1.12.0-rc4, build e4a0dbc, experimental
```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 1.12.0-rc4
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 9
 Dirperm1 Supported: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge overlay null host
Swarm: active
 NodeID: 8nqfybfvayeudhrm6vrzuuxws
 IsManager: Yes
 Managers: 1
 Nodes: 3
 CACertHash: sha256:6e33ad67f9239b7c8ea99648b8b3b1b6d5e28cd3120d9f197a06c70074cef544
Runtimes: runc
Default Runtime: runc
Security Options: apparmor
Kernel Version: 3.13.0-30-generic
Operating System: Ubuntu 14.04 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 337.4 MiB
Name: Zeuski
ID: O65Q:PDSP:WKFN:CWRF:CZUL:S6GV:H3IU:JOSU:WJWG:N3DP:A5Q3:LHAN
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Experimental: true
Insecure Registries:
 127.0.0.0/8
```

**Steps to reproduce the issue:**
1. `docker service create --name mongo mongo --replSet "rs0"`
2. `docker service update --arg "--replSet \"rs0\" --auth" mongo`

**Describe the results you received:**

`invalid argument "--replSet \"rs0\" --auth" for --arg: line 1, column 10: bare " in non-quoted-field`

**Describe the results you expected:**

The args would get updated and the tasks would be reset with the new args

**Additional information you deem important (e.g. issue happens only occasionally):**

I'm setting up a swarm with MongoDB. I need to update the service to use the "--auth" flag after I've enabled authentication on it. The documentation says that `service update` has an `--args` flag. This was false because when I ran `docker service update --help` it instead had a `--arg` flag. The value is totally opaque and there are no docs on what to do, so I guessed that I could just set a string, or an array like one would use in a dockerfile.

**Comment by dnephin** [comment]
This is already fixed on master. I tried it out, and ` docker service update --args "--replSet \"rs0\" --auth" mongo` works correctly.  `arg` was changed to `args` and supports the syntax you suspected. In RC4 is would have needed to be something like `--arg --replSet --arg rs0 --arg --ath`, but I'm not even sure if that would work.

It looks like the docs for this change were published before the change itself made it into an RC. The fix will be in the next RC.

FWIW, you don't need quotes around rs0, it should work as:

    docker service update --args "--replSet rs0 --auth" mongo
